### PR TITLE
helm: fix missing default resources in rook components

### LINKED ISSUE #10083: Helm Chart missing default resources for some provisioner sidecars
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Default resources are not supplied for all provisioner sidecars in the rook-ceph Helm Chart.

**Expected behavior:**
Default resources are set for all provisioner sidecars.

**How to reproduce it (minimal and precise):**
Grab the 1.9.0 release which introduces default resources and check the provisioner sidecars for missing resources.

It looks like it's currently missing a default for:
- the csi-snapshotter sidecar under csiCephFSProvisionerResource
- the csi-omap-generator

### PR DESCRIPTION
**Pull request by yuvalman**
Signed-off-by: Yuval Manor <yuvalman958@gmail.com>

**Description of your changes:**
1. Add csi-snapshotter resources for csiCephFSProvisioner
2. Add csi-omap-generator resources for csiRBDProvisioner (only if omap generator is enabled)

**Which issue is resolved by this Pull Request:**
Resolves #10083

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Review by travisn — COMMENTED** [review]
@Madhu-1 Does CSI have recommended resources for the omap generator and snapshotter?

File: deploy/charts/rook-ceph/values.yaml [hunk 1]
```diff
@@ -156,6 +157,14 @@ csi:
         limits:
           memory: 1Gi
           cpu: 500m
+    - name : csi-omap-generator
```
Comment (Madhu-1) [inline deploy/charts/rook-ceph/values.yaml:160]: Omap generator need to talk to ceph and it's cephcsi specific , use limit and request same as rbd-plugin

**Comment by Madhu-1** [comment]
> @Madhu-1 Does CSI have recommended resources for the omap generator and snapshotter?

@travisn missed notification , We can apply the rbdplugin resources to omap and for snapshotter we can apply same as other sidecar resources

**Review by Madhu-1 — APPROVED** [review]

**Comment by Madhu-1** [comment]
@yuvalman can you please make the same changes in operator.yaml and operator -openshift.yaml file?

**Comment by travisn** [comment]
> > @yuvalman can you please make the same changes in operator.yaml and operator -openshift.yaml file?
> 
> In operator.yaml and operator-openshift.yaml let's actually keep all the resources commented out, and just add the new resources commented out there. For now we just have enabled these default resources in the helm chart.

@yuvalman To be clear we do want change changes in operator.yaml and operator-openshift.yaml for these resources you're adding. We just want to keep them commented out in those files since we haven't enabled the resources by default in non-helm deployments. Does that make sense?

**Review by travisn — APPROVED** [review]

**Pull request merged**